File: frontend/src/model/FriendRequest.ts
export class FriendRequest {
    id: string;
    senderId: string;
    receiverId: string;
    senderUsername: string;
    accepted: boolean;
    requestDate: Date;

    constructor(id: string, senderId: string, receiverId: string, senderUsername: string, accepted: boolean, requestDate: Date) {
        this.id = id;
        this.senderId = senderId;
        this.receiverId = receiverId;
        this.senderUsername = senderUsername;
        this.accepted = accepted;
        this.requestDate = requestDate;
    }
}

File: backend-game-service/src/main/java/be/kdg/backendgameservice/domain/FriendRequest.java
package be.kdg.backendgameservice.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@Entity
public class FriendRequest {
    @Id
    @GeneratedValue
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sender_id", nullable = false)
    private Player sender;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "receiver_id", nullable = false)
    private Player receiver;


    private boolean accepted;

    private LocalDateTime requestDate;

    private LocalDateTime acceptedDate;
}
